src/gtc/passes/oir_optimizations/horizontal_execution_merging.py

```diff
@@ -199,12 +236,13 @@ def visit_VerticalLoop(self, node: oir.VerticalLoop, **kwargs: Any) -> oir.Verti
     def visit_Stencil(self, node: oir.Stencil, **kwargs: Any) -> oir.Stencil:
         vertical_loops: List[oir.VerticalLoop] = []
         protected_fields = set(n.name for n in node.params)
+        all_names = collect_symbol_names(node)
         for vl in reversed(node.vertical_loops):
```

Comment:
I don't know if it is actually faster, but this for loop can be easily rewritten as a list comprehension and it's (maybe ?) simpler to understand

```python
vertical_loops = [
    *reversed(
        self.visit(
            vl,
            new_symbol_name=symbol_name_creator(all_names),
            protected_fields=protected_fields,
            **kwargs,
        )
        for vl in reversed(node.vertical_loops)
    )
]
```

Reply:
It turns out that `reversed` requires a Sequence or at least Reversible type (which the generator is not) so I had to make an inner list as well, but it should work.

Reply:
I realized my suggestion was completely wrong and it should be reverted. First, it requires creating an extra temporary list for `reversed` to work as @jdahm mentioned, and second, and more importantly, the other two statements inside the loop have been removed in the list comprehension:
```
access_collection = AccessCollector.apply(vl)
protected_fields |= access_collection.fields()
```
The funny thing is that tests are still passing even without these two statements, so I think the tests should be improved to catch this error.